ts/shielder-sdk-crypto-wasm-light/src/circuits/crypto.ts

```diff
@@ -0,0 +1,97 @@
+import { uint8ToHex } from "@/nitro-attestation";
+import { hexToUint8 } from "@/utils";
+import * as secp from "@noble/secp256k1";
+
+type Keypair = { sk: Uint8Array; pk: Uint8Array };
+
+export function generateKeypair(): Keypair {
+  const sk = secp.utils.randomPrivateKey();
+  const pk = secp.getPublicKey(sk, true);
+  return { sk, pk };
+}
+
+async function hkdf(
+  sharedSecret: Uint8Array,
+  cryptoAPI: Crypto
+): Promise<CryptoKey> {
+  const keyMaterial = await cryptoAPI.subtle.importKey(
+    "raw",
+    sharedSecret,
+    "HKDF",
+    false,
+    ["deriveKey"]
+  );
+  return cryptoAPI.subtle.deriveKey(
+    {
+      name: "HKDF",
+      hash: "SHA-256",
+      salt: new Uint8Array([]),
+      info: new TextEncoder().encode("ecies-secp256k1-v1")
+    },
+    keyMaterial,
+    { name: "AES-GCM", length: 256 },
+    false,
+    ["encrypt", "decrypt"]
+  );
+}
+
+export async function encrypt(
```

Comment:
If this code was taken from https://github.com/Cardinal-Cryptography/ecies-encryption-poc/blob/main/js/index.ts , can we extract this to a separate TS lib? Can be done in the next PRs

Reply:
Right, let's create a task for now, I'll resolve this conv